net/nimble/host/src/ble_store.c

```diff
@@ -28,49 +28,98 @@ ble_store_read(int obj_type, const union ble_store_key *key,
 {
     int rc;
 
+    ble_hs_lock();
```

Comment:
You're right - I snuck it into this PR :).  I do think we need these locks here to prevent concurrent reads and writes.  For example, if code is iterating the RAM store while a record is being written, a garbage record could be retrieved.